app/interactors/show_user.rb

```diff
@@ -0,0 +1,25 @@
+class ShowUser
+  include Interactor
+
+  def call
+    validate_input
+    execute
+    validate_output
+  end
+
+  private
+
+  def validate_input
+    context.fail!(error: "invalid user id") unless context.id
+  end
+
+  def execute
+    context.user = User.find(context.id)
+  rescue ActiveRecord::RecordNotFound
+    context.user = nil
+  end
+
+  def validate_output
+    context.fail!(error: "invalid user") unless context.user
```

Comment:
@Enriikke do you like to handle errors differently? It seems that most instances where I'd want to display errors it's due to a lack of an ActiveRecord model (so I usually can't pull from `User.errors` for example).